devenv/lib/archive.py

```diff
@@ -20,17 +21,34 @@ def atomic_replace(src: str, dest: str) -> None:
     os.replace(src, dest)
 
 
-def download(url: str, sha256: str, dest: str = "") -> str:
+def download(
+    url: str,
+    sha256: str,
+    dest: str = "",
+    retries: int = 3,
+    retry_exp: float = 2.0,
+) -> str:
+    assert retries >= 0
```

Comment:
```suggestion
    if retries < 0:
        raise ValueError("Retries cannot be negative")
```
Since assertions can be disabled, I feel an explicit check would be better

Reply:
good one!